dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/relation/RelationModelManager.java

```diff
@@ -0,0 +1,596 @@
+package com.dat3m.dartagnan.verification.model.relation;
+
+import com.dat3m.dartagnan.encoding.EncodingContext;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.event.core.RMWStore;
+import com.dat3m.dartagnan.program.filter.Filter;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.solver.caat.predicates.CAATPredicate;
+import com.dat3m.dartagnan.solver.caat.predicates.PredicateHierarchy;
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.base.SimpleGraph;
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.derived.*;
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.Edge;
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.RelationGraph;
+import com.dat3m.dartagnan.solver.caat4wmm.EventDomainNext;
+import com.dat3m.dartagnan.utils.dependable.DependencyGraph;
+import com.dat3m.dartagnan.verification.model.ExecutionModelManager;
+import com.dat3m.dartagnan.verification.model.ExecutionModelNext;
+import com.dat3m.dartagnan.verification.model.event.*;
+import com.dat3m.dartagnan.verification.model.relation.RelationModel.EdgeModel;
+import com.dat3m.dartagnan.verification.model.ThreadModel;
+import com.dat3m.dartagnan.wmm.analysis.RelationAnalysis;
+import com.dat3m.dartagnan.wmm.Constraint.Visitor;
+import com.dat3m.dartagnan.wmm.definition.*;
+import com.dat3m.dartagnan.wmm.Relation;
+import com.dat3m.dartagnan.wmm.Wmm;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.Lists;
+import org.sosy_lab.common.configuration.Option;
+import org.sosy_lab.common.configuration.Options;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+
+import java.math.BigInteger;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.dat3m.dartagnan.configuration.OptionNames.WITNESS_RELATIONS_TO_SHOW;
+import static com.dat3m.dartagnan.wmm.RelationNameRepository.*;
+
+@Options
+public class RelationModelManager {
+
+    private static final Logger logger = LogManager.getLogger(RelationModelManager.class);
+
+    private final ExecutionModelManager manager;
+    private final RelationGraphBuilder graphBuilder;
+    private final RelationGraphPopulator graphPopulator;
+
+    private final Map<Relation, RelationModel> relModelCache;
+    private final BiMap<Relation, RelationGraph> relGraphCache;
+    private final Map<String, EdgeModel> edgeModelCache;
+
+    private EncodingContext context;
+    private Wmm wmm;
+    private ExecutionModelNext executionModel;
+    private EventDomainNext domain;
+
+    @Option(name=WITNESS_RELATIONS_TO_SHOW,
+            description="Names of relations to show in the witness graph.",
+            secure=true)
+    private String relToShowStr = "default";
```

Comment:
Haven't tried yet. But this is a question about whether we should: 1) process the input in a different way (using `String`) but let users have a consistent command style `--options=a,b,c` or 2) process the input in a near-consistent way (`Set` but not `EnumSet`) but users have to type this option in a different style, although we can document properly. I would pick the former but will still modify it if you are sticking with the latter.

Reply:
I have conflicting opinions here: as a user I would prefer to have consistent format for the options. As a maintainer/developer I would want to have a consistent implementation.

@ThomasHaas what do you think?

Reply:
What is the problem with using regular strings and setting the default to `"po, rf, co"`? Is it even reasonable to not draw po and rf? Especially without po, your graph is likely unreadable because it won't resemble the program code in the slightest.

Also, I personally don't mind using sets, even if the user has to put braces `{...}`.

Reply:
There is no problem with which relations are shown by default. We are just talking about how to process the input of this option typed by users: with String or Set?

Reply:
I understand. I don't mind either way, but if you use strings then `"default"` is not the correct default string, and neither is the empty string `""` as suggested by @hernanponcedeleon (unless some core relations like `po` are always drawn).

Reply:
Are you saying that po, co, rf are always shown, or that they are shown by default? That is still not clear to me and my point of discussion. If empty string means "show po, co, rf" and any other value means "just show whatever I specified", then I'm saying that this is just wrong and the default value should not have been empty string but simply "po, co, rf".

Put differently, is it possible for the user to somehow not show one of the core relations, e.g. `po`, by specifying `--options=rf,co`?
Just to be precise: this has nothing to do with String vs. Set as I do not mind either way.

Reply:
I mean: if user types nothing for this option, i.e. the empty string, po, co, rf will be shown. When user types something, we show relations as he wants. 

>is it possible for the user to somehow not show one of the core relations, e.g. po, by specifying just --options=rf,co?

Of course yes. Just type `--witness.show=rf,co`.